docs/design/resource-states.md

```diff
@@ -0,0 +1,222 @@
+   positive polarity conditions like `Ready`, which as mentioned above is a pattern followed by many operators. This _works_ with kstatus but [isn't as ideal](https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/README.md#the-ready-condition).
+   Possibly in order to dodge some of the issues which kstatus has with positive polarity conditions we could add the `Ready` condition as false via a webhook when a resource is added.
+
+## An aside on ARM templates
+This isn't related to the core topic, but it might be useful to understand because it has a big impact on where failure is possible when communicating with ARM.
+
+You can read some guidelines about how ARM templates work [here](https://armwiki.azurewebsites.net/api_contracts/guidelines/templatedeployment.html).
+
+The process basically boils down to the following and is documented [here](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md):
+1. ARM: Validate basic template syntax. Return an error for invalid JSON, invalid resource ID structure, etc.
+2. ARM: Send the resource to the individual resource provider (RP).
+3. RP: Return a response for the resource. This can be:
+   * `201` or `200` HTTP status code and a response body. At this point the resource _should_ exist, but with possibly with a non-terminal `provisioningState`.
+   * `202` HTTP status code. At this point the resource _should not_ exist, and will not exist until the long-running operation completes successfully.
+4. ARM: Poll long running operation until it completes (if needed).
+5. ARM: Deployment status will be the status of the resource at the end of the long running operation, which should be in one of the 3 terminal states `Succeeded`, `Failed`, or `Canceled`
+
+Possible outcomes:
+
+|      Status of deployment                                                  |               State of resource                                                                                   |
+|----------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| 4xx HTTP error code when creating deployment                               | Resource does not exist                                                                                           |
+| Deployment accepted, but fails after polling                               | Resource may or may not exist (depends on the RP)                                                                 |
+| Deployment succeeds                                                        | Resource has `Failed` `provisioningState` - The spec doesn't say this is impossible but it's probably very rare   |
+| Deployment succeeds                                                        | Resource has `Succeeded` `provisioningState`                                                                      |
```

Comment:
I use a Visual Studio Code plugin that will reformat tables for clarity. See [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one)

```suggestion
| Status of deployment                         | State of resource                                                                                               |
| -------------------------------------------- | --------------------------------------------------------------------------------------------------------------- |
| 4xx HTTP error code when creating deployment | Resource does not exist                                                                                         |
| Deployment accepted, but fails after polling | Resource may or may not exist (depends on the RP)                                                               |
| Deployment succeeds                          | Resource has `Failed` `provisioningState` - The spec doesn't say this is impossible but it's probably very rare |
| Deployment succeeds                          | Resource has `Succeeded` `provisioningState`                                                                    |

```

Reply:
Fixed